core/jvm/src/main/scala/cats/effect/unsafe/metrics/WorkStealingThreadPoolMetrics.scala

```diff
@@ -85,12 +85,54 @@ sealed trait WorkStealingThreadPoolMetrics {
    */
   def suspendedFiberCount(): Long
 
+  /**
+   * External queue metrics for this work-stealing thread pool.
+   */
+  def externalQueue: ExternalQueueMetrics
+
   /**
    * The list of worker-specific metrics of this work-stealing thread pool.
    */
   def workerThreads: List[WorkerThreadMetrics]
 }
 
+/**
+ * Represents metrics for the external task queue in a work-stealing thread pool.
+ */
+sealed trait ExternalQueueMetrics {
+
+  /**
+   * Returns the total number of singleton tasks submitted to the queue.
+   */
+  def singletonsSubmittedCount(): Long
+
+  /**
+   * Returns the number of singleton tasks currently in the queue.
+   */
+  def singletonsPresentCount(): Long
+
+  /**
+   * Returns the total number of batch tasks submitted to the queue.
+   */
+  def batchesSubmittedCount(): Long
+
+  /**
+   * Returns the number of batch tasks currently in the queue.
+   */
+  def batchesPresentCount(): Long
```

Comment:
This matches our existing naming convention. Also, I realized we should add two more metrics: `fiberCount` and `totalFiberCount`, that counts singletons and the number of fibers in the batches.

```suggestion
  def totalSingletonCount(): Long

  /**
   * Returns the number of singleton tasks currently in the queue.
   */
  def singletonCount(): Long

  /**
   * Returns the total number of batch tasks submitted to the queue.
   */
  def totalBatchCount(): Long

  /**
   * Returns the number of batch tasks currently in the queue.
   */
  def batchCount(): Long
```

Reply:
Okay, for sure!